### PR TITLE
Dolev/add tests for rules

### PR DESCRIPTION
**Pull request by dolby360**
@dimabru 
As described [here](https://github.com/allero-io/allero/issues/53) this pr is the infrastructure to test whether rules are passed or failed for certain schemas.
For now, only support github and rule 1. But the infra is built to easily add more scm platforms and rules.

### CODE REVIEW

**Comment by dimabru** [comment]
@dolby360 
Looks like you have some compilation issues.
I'll review once they're fixed.

Let me know if you need help with it!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,3 +25,6 @@ jobs:
     - name: Run vet
       run: |
         go vet .
+
+    - name: Test rules
+      run: go test -v ./pkg/rulesConfig/rules_test.go
```
Comment (dimabru) [inline .github/workflows/ci.yml:30]: missing some newlines
I suggest using the following extension for vscode: samverschueren.final-newline

**Review by dimabru — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dolby360) [reply .github/workflows/ci.yml:30]: Done:)

**Comment by dolby360** [comment]
@dimabru Let's wait with the merge at least until I will add all github test files :)

**Comment by dolby360** [comment]
@dimabru OK I finished with github tests, besides 5, 10, 11, 14, 15 , 16 , 17. Which I'll leave for another time.
I thought maybe even start with gitlab tests here too, but it is too much. after this PR will be merged I'll open a new issue for github tests and I'll work on them.

**Review by dimabru — APPROVED** [review]

**Comment by dimabru** [comment]
@dolby360 Thanks a lot for all the work!!
I made some small changes and pushed to your branch 

fixed typo `faile` to `fail` in test filenames, added a step to the release flow to make sure tests passed and a few other small tweaks

**Pull request merged**